force-app/main/arena/classes/KafkaArenaHandler_Helper.cls

```diff
@@ -128,6 +128,15 @@ public inherited sharing class KafkaArenaHandler_Helper {
                 return null;
             }
 
+            if (d.year() > 4000) {
+                Integer day = d.day();
+                Integer month = d.month();
+                Integer year = 4000;
+                Date newDate = date.newInstance(year, month, day);
+
+                return newDate;
```

Comment:
```suggestion
                return null;
```

Synes ikke vi burde endre dato. Er den for høy for Salesforce, så kommer den bare ikke med. Da kan du også slå sammen de to if-ene, til én if-setning.